hazelcast/src/main/java/com/hazelcast/util/IterableUtil.java

```diff
@@ -71,4 +72,46 @@ public void remove() {
     public static <T> Iterable<T> nullToEmpty(Iterable<T> iterable) {
         return iterable == null ? Collections.<T>emptyList() : iterable;
     }
+
+    public static <T> Iterable<T> createIterableWithSingleItem(final T item) {
+        return new Iterable<T>() {
+            @Override
+            public Iterator<T> iterator() {
+                return new SingleItemIterator(item);
+            }
+        };
+    }
+
+    /**
+     * utility class to create a single-item iterator
+     */
+    private static class SingleItemIterator<T> implements Iterator<T> {
```

Comment:
What's the purpose of `final` modifier here? It's already not visible out of `IterableUtil`...